src/render_tree.rs

```diff
@@ -122,7 +126,22 @@ impl RenderTree {
                     NodeData::Text(text) => {
                         let mut mut_element_ref = reference_element.as_ref().borrow_mut();
                         if let NodeType::Text(element_text) = &mut mut_element_ref.node_type {
-                            element_text.value.push_str(text.value());
+                            unsafe {
+                                let try_realloc = libc::realloc(
+                                    element_text.value as *mut c_void,
+                                    libc::strlen(element_text.value)
+                                        + text.value().len()
+                                        + 1 as size_t,
+                                );
+                                if !try_realloc.is_null() {
+                                    element_text.value = try_realloc as *mut c_char;
+                                    libc::strncat(
+                                        element_text.value,
+                                        text.value().as_ptr() as *const c_char,
+                                        text.value().len() as size_t,
+                                    );
+                                }
+                            }
```

Comment:
this is the main piece of "unsafe" code in the engine (outside of the bindings themselves)

We can have multiple text nodes tied to a node, these need to be "concatenated" together to create the text body of that node. Since the underlying type is a raw C ptr, we have to use some libc functions that (attempt to) rellocate the appropriate amount of memory for the new string and call `strncat` to append the text (with one extra byte for the null terminator)

if the reallocation fails, this exits early and the additional text nodes (in theory) should be ignored

Reply:
No longer relevant, unsafe code has been removed from the engine